### ISSUE TITLE
Application cache is not clean up and consumes 64MB

### ISSUE THREAD

**Issue by marulinko** [TeamNewPipe/NewPipe#662]
After playing video / audio using new pipe is application cache utilized to 64MB and this cache is never clean up and must be done manually using android menu.
Would be possible to implement application cache cleanup after exit or after some trigger?
I mean these files 
org.schabi.newpipe/cache/exoplayer/*.v3.exo
each file size is 524.29KB

**Comment by theScrabi** [comment]
Actually that may not sound like such a bad Idea. @mauriciocolli @TheAssassin what do you think?

**Comment by mauriciocolli** [comment]
I don't think we could safely clean the cache as we can not predict when the files won't be used again (although the user probably won't watch the video over and over again everyday).
Taking a browser for example, it just evict the older data (or less used) to make room for newer one.

But the point is, it doesn't make sense to clean the cache as when you watch another video, it'll fill all up again, so you can treat the space as "reserved".

I see two "solutions" for this problem:
- Let the user specify the cache size (like some browsers)
- Extending the first, let the user disable the cache entirely 

The latter brings some problems like: when you switch players (popup and normal player) or to seeking to a previous time.

**TL;DR:** We can decide when to clean the cache as it could mess up with seeking (and some other features) and it doesn't matter that we clean it, it'll fill up again.

**Comment by TheAssassin** [comment]
Well, those are caches, aren't they? I assume they attempt to make it easier if not possible at all for exoplayer to buffer the video, and exoplayer might even reuse the data on subsequent playbacks.

I don't think it's safe to assume that these files can be deleted. But I'd suggest to add a button in the settings cleaning them up. Then, no video is played back, it should be somewhat safe to clean them up.

**Comment by theScrabi** [comment]
@TheAssassin Although if you take it precise. You can enter settings while overlay player is playing, so...
